jbpm-flow/src/main/java/org/jbpm/workflow/instance/node/CompositeNodeInstance.java

```diff
@@ -262,7 +271,7 @@ public NodeInstance getNodeInstance(final Node node) {
         // async continuation handling
         if (node instanceof AsyncEventNode) {
             actualNode = ((AsyncEventNode) node).getActualNode();
-        } else if (useAsync(node)) {
+        } else if (useAsync(node) && wrapAsync) {
```

Comment:
why do we need this wrapAsync?

Reply:
that one is tricky
https://github.com/kiegroup/jbpm/pull/1465/files#diff-88f436b26f11134f1919b7fe8c0f4b38R162

if you try to get the node instance from an async node you will get an async node because of the env entry AsyncMode = true... I needed a way to avoid that.

Reply:
@mswiderski  ^^

Reply:
so why not putting this logic into useAsync? So if it is already async node it should not wrap it or did I misunderstand it?

Reply:
that is fine by me. you are just moving some logic around.